src/index.ts

```diff
@@ -1,50 +1,50 @@
-import net from 'net'
+import net from 'net';
 
 export declare interface Status {
-  ping: number
-  version: string
-  motd: string
-  players: number
-  max_players: number
+  ping: number;
+  version: string;
+  motd: string;
+  players: number;
+  max_players: number;
 }
 
 export declare interface McServer {
-  host: string
-  port: number
+  host: string;
+  port: number;
 }
 
 export function checkStatus(server: McServer): Promise<Status> {
 
   return new Promise((resolve, reject) => {
-    const start_time = new Date()
-    let ping: Number
-    let data: string
+    const start_time = new Date();
+    let ping: Number;
+    let data: string;
 
     const client = net.connect(server, () => {
-      ping = Math.round(new Date().getMilliseconds() - start_time.getMilliseconds())
+      ping = Math.round(new Date().getMilliseconds() - start_time.getMilliseconds());
 
-      let buff = Buffer.from([ 0xFE, 0x01 ])
-      client.write(buff)
-    })
+      let buff = Buffer.from([0xFE, 0x01]);
+      client.write(buff);
+    });
 
     client.on('data', (d) => {
-      data = d.toString()
-      client.destroy()
-    })
+      data = d.toString();
+      client.destroy();
+    });
 
     client.on('close', () => {
-      let server_info = data.split('\x00\x00\x00')
+      let server_info = data?.split('\x00\x00\x00');
```

Comment:
Why not return early if server_info is unavailable?

Reply:
I like that idea. I don't like semicolons. Resubmit without them and I'll merge it in